app/actions/utils.js

```diff
@@ -56,10 +60,35 @@ export function makeDisplayModal(dispatch) {
   };
 }
 
-export function makeUploadCb(dispatch, getState, errCode, utc) {
-  return (err, recs) => {
+export function makeUploadCb(dispatch, getState, errCode, utc, api) {
+  return async (err, recs) => {
```

Comment:
This will be the first `async` function outside of device code. I'm not opposed to using the newer syntax, but it does bring up a slightly larger conversation around consistency.

Reply:
And I think this will be less relevant if the `api` signature is changed to accept a callback instead of return a `Promise`

Reply:
I thought we could use any ES6 functions, but I understand you concern regarding consistency and confusion about it being `Promise` or a callback that's being returned. Will change to callbacks.

Reply:
We can use any ES6 functions. We've just not made the conceptual jump to a newer async control semantics in any of our repos yet. While I very much enjoy the async/await syntax I think that moving to it makes more sense as a larger coordinated changeset as opposed to just migrating one function at a time.